include/cspublic.h

```diff
@@ -770,10 +770,15 @@ CS_RETCODE cs_ctx_drop(CS_CONTEXT * ctx);
 CS_RETCODE cs_config(CS_CONTEXT * ctx, CS_INT action, CS_INT property, CS_VOID * buffer, CS_INT buflen, CS_INT * outlen);
 CS_RETCODE cs_strbuild(CS_CONTEXT * ctx, CS_CHAR * buffer, CS_INT buflen, CS_INT * resultlen, CS_CHAR * text, CS_INT textlen,
 		       CS_CHAR * formats, CS_INT formatlen, ...);
-#undef cs_dt_crack
+#ifndef cs_dt_crack_v2
+#  undef cs_dt_crack
+#endif
 CS_RETCODE cs_dt_crack(CS_CONTEXT * ctx, CS_INT datetype, CS_VOID * dateval, CS_DATEREC * daterec);
 CS_RETCODE cs_dt_crack_v2(CS_CONTEXT * ctx, CS_INT datetype, CS_VOID * dateval, CS_DATEREC * daterec);
-#define cs_dt_crack cs_dt_crack_v2
+#ifndef _FREETDS_LIBRARY_SOURCE
+#  undef cs_dt_crack
+#  define cs_dt_crack cs_dt_crack_v2
+#endif
```

Comment:
I prefer the previous style. Shorter and does not require adding a new `_FREETDS_LIBRARY_SOURCE` definition.

Reply:
Understood, but it interacts poorly with mass renaming (e.g., tacking on `_ftds14` suffixes across the board) that would otherwise allow coexistence with commercial MS/SAP libraries and/or other analogously patched FreeTDS versions.  I'm obviously not proposing to push any such renaming, just the changes that make it fully possible.

Reply:
I don't understand the need for mass renaming. I don't think it's possible to use both proprietary library and FreeTDS without breaking ABI (Windows and ODBC excluded), the 2 libraries will clash.
Usually to allow this usually libraries use symbol prefixes instead of suffix (for instance adding `ftds_` before all exported functions, I know `libiconv` is adding `lib` prefix to functions).

Reply:
Yeah, this is a special case -- the C++ Toolkit performs mass renaming for the FreeTDS versions it bundles, with each version's wrapper in a separate namespace but all wrappers implementing the same base classes.

Reply:
Why not using dynamic linking (and loading) or different versions?

Reply:
To accommodate binaries that link non-system libraries statically for ease of deployment.

Reply:
Thinking about this and `dbopen` and `cs_dt_crack`. They are provided to keep ABI compatibility. But if you link statically code should not use them at all. So in case of static building they should not be compiled or even declared.
If I understood how this renaming is done you have a lot of different defines like
```c
#define dbconvert dbconvert_ftds15
#define dbconvert_ps dbconvert_ps_ftds15
...
```
so you want to avoid the `#undef`s. Is that right?